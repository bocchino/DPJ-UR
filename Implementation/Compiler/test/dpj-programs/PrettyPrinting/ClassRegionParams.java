/**
 * Basic class region parameter
 */
class C1<region P> {
    region r;
    C1<r> field;
    void m1() {
	field = new C1<r>();
	C1<P> local = new C1<P>();
    }
}

/**
 * Class with region parameter and type parameter
 */
class C2<type T, region P> {
    region r;
    C2<T,r> field;
    void m1() {
        field = new C2<T,r>();
        C2<T,P> local = new C2<T,P>();
    }
}

/**
 * Two class region parameters, one declared under an RPL
 */
class C3<region P1, P2> {
    region r1;
}

/**
 * Two class region parameters, one declared under an RPL and 
 * both declared disjoint
 */
class C4<region P1, P2 in C4.r1:* | P1 # P2> {
    region r1;
}

/**
 * One class region parameter shared with the superclass
 */
class C5<region P,Q> extends C1<Q> {
    region s;
    C5<r,s> anotherField;
}