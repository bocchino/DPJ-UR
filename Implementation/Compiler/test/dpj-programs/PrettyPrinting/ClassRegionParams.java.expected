class C1<region P> {
    
    C1() {
        super();
    }
    region r;
    C1<r> field;
    
    void m1() {
        field = new C1<r>();
        C1<P> local = new C1<P>();
    }
}
class C2<T, region P> {
    
    C2() {
        super();
    }
    region r;
    C2<T, r> field;
    
    void m1() {
        field = new C2<T, r>();
        C2<T, P> local = new C2<T, P>();
    }
}
class C3<region P1, P2> {
    
    C3() {
        super();
    }
    region r1;
}
class C4<region P1, P2 in C4.r1:* | P1 # P2> {
    
    C4() {
        super();
    }
    region r1;
}